src/Components/Components/src/ComponentsMetrics.cs

```diff
@@ -0,0 +1,239 @@
+            description: "Total number of exceptions during browser event processing.");
+
+        _parametersDuration = _lifeCycleMeter.CreateHistogram(
+            "aspnetcore.components.update_parameters.duration",
+            unit: "s",
+            description: "Duration of processing component parameters.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _parametersException = _lifeCycleMeter.CreateCounter<long>(
+            "aspnetcore.components.update_parameters.exceptions",
+            unit: "{exception}",
+            description: "Total number of exceptions during processing component parameters.");
+
+        _batchDuration = _lifeCycleMeter.CreateHistogram(
+            "aspnetcore.components.rendering.batch.duration",
+            unit: "s",
+            description: "Duration of rendering batch.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _batchException = _lifeCycleMeter.CreateCounter<long>(
+            "aspnetcore.components.rendering.batch.exceptions",
+            unit: "{exception}",
+            description: "Total number of exceptions during batch rendering.");
+    }
+
+    public void Navigation(string componentType, string route)
+    {
+        var tags = new TagList
+        {
+            { "component.type", componentType ?? "unknown" },
+            { "route", route ?? "unknown" },
+        };
+
+        _navigationCount.Add(1, tags);
+    }
+
+    public async Task CaptureEventDurationAsync(Task task, long startTimestamp, string? componentType, string? methodName, string? attributeName)
+    {
+        try
+        {
+            await task;
+
+            var tags = new TagList
+            {
+                { "component.type", componentType ?? "unknown" },
+                { "component.method", methodName ?? "unknown" },
+                { "attribute.name", attributeName ?? "unknown" }
+            };
+
+            var duration = Stopwatch.GetElapsedTime(startTimestamp);
+            _eventDuration.Record(duration.TotalSeconds, tags);
+        }
+        catch
+        {
+            // none
+        }
+    }
+
+    public async Task CaptureParametersDurationAsync(Task task, long startTimestamp, string? componentType)
+    {
+        try
+        {
+            await task;
+
+            var tags = new TagList
+            {
+                { "component.type", componentType ?? "unknown" },
+            };
+
+            var duration = Stopwatch.GetElapsedTime(startTimestamp);
+            _parametersDuration.Record(duration.TotalSeconds, tags);
+        }
+        catch
+        {
+            // none
+        }
+    }
+
+    public void BatchDuration(long startTimestamp, int diffLength)
+    {
+        var tags = new TagList
+        {
+            { "diff.length.bucket", BucketEditLength(diffLength) }
```

Comment:
`diff.approx_length`? I was a bit confused by the bucket in the name. Totally the right thing to do to bucketize the sizes as that eliminates dimension explosion, but the name confused me as to whether the feature has buckets or something)

@lmolkova - do you know of a precedent for where this is done elsewhere?

Reply:
made it `diff.approximate.length`, happy to change it further